### PR TITLE
Set test timeout to 30 mins

### PR DESCRIPTION
**Pull request by mhdawson**
Shorten the timeout from the default of 360 minutes.

### CODE REVIEW

**Comment by mhdawson** [comment]
@cjihrig this will help in seeing if any changes that I've made are helping the windows/macos CIs.

**Review by cjihrig — APPROVED** [review]
LGTM. We can probably go even shorter, but this is an improvement.

**Comment by mhdawson** [comment]
@cjihrig what time do you think makes sense?  Happy to make it shorter.

**Comment by cjihrig** [comment]
I think 10 minutes should be more than enough for runs that are going to pass.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ jobs:
   build:
     name: ${{ matrix.os }} - ${{ matrix.config.variation }}
     runs-on: ${{ matrix.os }}
+    timeout-minutes: 30
```
Comment (mhdawson) [inline .github/workflows/CI.yml:9]:
```suggestion
    timeout-minutes: 10
```

**Comment by mhdawson** [comment]
@cjihrig updated to be 10 mins. I'll land once the CI's complete.

**Pull request merged**